src/Deedle/DelayedSeries.fs

```diff
@@ -143,15 +142,18 @@ type internal DelayedSource<'K, 'V when 'K : equality>
     let ranges = flattenRanges rangeMin rangeMax comparer ranges |> Array.ofSeq 
     let ops = loader ranges
     async {
-      let data = new ResizeArray<_>(1000)
+      let constrs = ResizeArray<_>()
+      let vectors = ResizeArray<_>()
       for ((lo, lob), (hi, hib)), op in Seq.zip ranges ops do
-        let! rangeValues = op
-        for KeyValue(k, v) in rangeValues do
-          if (k > lo || (k >= lo && lob = Inclusive)) &&
-             (k < hi || (k <= hi && hib = Inclusive)) then data.Add( (k, v) )
-      let vector = vectorBuilder.Create(data.ToArray() |> Array.map snd)
-      let index = indexBuilder.Create(Seq.map fst data, Some true)
-      return index, vector } |> Async.StartAsTask)
+        // TODO: We used to fix boundary conditions if the loader was wrong
```

Comment:
We still do this (below) for the loader that returns a sequence of key value pairs. But we do not "fix" incorrect results for the new API - I'll see if we need to do this or not.